gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/mapper/PlanMapper.java

```diff
@@ -158,15 +157,38 @@ private Duration calculateDuration(TimePeriodConfiguration configuration) {
             .dividedBy(configuration.getLimit().intValue());
     }
 
-    private long getLimitToUse(JsonNode configuration) {
+    private long getLimitToUse(JsonNode configuration, GenericApiEntity api) {
         // Dynamic limit exists and limit is either missing or equal to 0
         if (
             configuration.has(POLICY_CONFIGURATION_DYNAMIC_LIMIT) &&
             (!configuration.has(POLICY_CONFIGURATION_LIMIT) || configuration.get(POLICY_CONFIGURATION_LIMIT).intValue() == 0)
         ) {
-            return Long.parseLong(configuration.get(POLICY_CONFIGURATION_DYNAMIC_LIMIT).asText());
+            return parseLimit(configuration.get(POLICY_CONFIGURATION_DYNAMIC_LIMIT), api);
+        }
+        return parseLimit(configuration.get(POLICY_CONFIGURATION_LIMIT), api);
+    }
+
+    private long parseLimit(JsonNode limitNode, GenericApiEntity api) {
+        var evaluatedLimit = evaluateLimit(limitNode, api);
+        if (evaluatedLimit != null) {
+            try {
+                return Long.parseLong(evaluatedLimit);
+            } catch (NumberFormatException ignored) {}
+        }
+        return 0L;
```

Comment:
I was not sure about this 0 default value when I have read the code. Have we tested how it behave when the limit is 0 in case of error ?

Reply:
In the Gateway, when the limit is 0 or not defined and the dynamic limit references via EL a property that does not exist, the endpoint is in HTTP Status 500 error (there is a NPE in the Gateway). 

It seems that the limit is not necessarily applied if everything is at 0, however, the endpoint doesn't work either.

Even when there is no dynamic limit used and the limit is at 0, I get this error in the Gateway:
```txt
java.lang.Exception: An error occurred while trying to execute policy rate-limit
	at io.gravitee.gateway.reactive.policy.adapter.policy.PolicyAdapter.lambda$policyExecute$0(PolicyAdapter.java:104)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableCreate.subscribeActual(CompletableCreate.java:40)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableDoFinally.subscribeActual(CompletableDoFinally.java:43)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapCompletable$ConcatMapCompletableObserver.drain(FlowableConcatMapCompletable.java:197)
	at io.reactivex.rxjava3.internal.operators.mixed.ConcatMapXMainSubscriber.onSubscribe(ConcatMapXMainSubscriber.java:77)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableFromIterable.subscribe(FlowableFromIterable.java:69)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableFromIterable.subscribeActual(FlowableFromIterable.java:47)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16144)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapCompletable.subscribeActual(FlowableConcatMapCompletable.java:56)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapCompletable$ConcatMapCompletableObserver.drain(FlowableConcatMapCompletable.java:197)
	at io.reactivex.rxjava3.internal.operators.mixed.ConcatMapXMainSubscriber.onNext(ConcatMapXMainSubscriber.java:106)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableDoOnEach$DoOnEachSubscriber.onNext(FlowableDoOnEach.java:92)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableSwitchIfEmpty$SwitchIfEmptySubscriber.onNext(FlowableSwitchIfEmpty.java:59)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableCache.replay(FlowableCache.java:272)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableCache.onNext(FlowableCache.java:317)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapMaybe$ConcatMapMaybeSubscriber.drain(FlowableConcatMapMaybe.java:238)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapMaybe$ConcatMapMaybeSubscriber.request(FlowableConcatMapMaybe.java:108)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableCache.onSubscribe(FlowableCache.java:298)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapMaybe$ConcatMapMaybeSubscriber.onSubscribeDownstream(FlowableConcatMapMaybe.java:102)
	at io.reactivex.rxjava3.internal.operators.mixed.ConcatMapXMainSubscriber.onSubscribe(ConcatMapXMainSubscriber.java:75)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableFromIterable.subscribe(FlowableFromIterable.java:69)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableFromIterable.subscribeActual(FlowableFromIterable.java:47)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16144)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapMaybe.subscribeActual(FlowableConcatMapMaybe.java:59)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16144)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableCache.subscribeActual(FlowableCache.java:115)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16144)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16090)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableDefer.subscribeActual(FlowableDefer.java:42)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16144)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableSwitchIfEmpty.subscribeActual(FlowableSwitchIfEmpty.java:32)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16144)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableDoOnEach.subscribeActual(FlowableDoOnEach.java:50)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16144)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapCompletable.subscribeActual(FlowableConcatMapCompletable.java:56)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable$SourceObserver.onComplete(CompletableAndThenCompletable.java:67)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable$NextObserver.onComplete(CompletableAndThenCompletable.java:99)
	at io.reactivex.rxjava3.internal.disposables.EmptyDisposable.complete(EmptyDisposable.java:68)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableEmpty.subscribeActual(CompletableEmpty.java:27)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableDefer.subscribeActual(CompletableDefer.java:43)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable$SourceObserver.onComplete(CompletableAndThenCompletable.java:67)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable$NextObserver.onComplete(CompletableAndThenCompletable.java:99)
	at io.reactivex.rxjava3.internal.util.AtomicThrowable.tryTerminateConsumer(AtomicThrowable.java:154)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapCompletable$ConcatMapCompletableObserver.drain(FlowableConcatMapCompletable.java:166)
	at io.reactivex.rxjava3.internal.operators.mixed.ConcatMapXMainSubscriber.onSubscribe(ConcatMapXMainSubscriber.java:77)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableFromIterable.subscribe(FlowableFromIterable.java:69)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableFromIterable.subscribeActual(FlowableFromIterable.java:47)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16144)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapCompletable.subscribeActual(FlowableConcatMapCompletable.java:56)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable$SourceObserver.onComplete(CompletableAndThenCompletable.java:67)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable$NextObserver.onComplete(CompletableAndThenCompletable.java:99)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableDoFinally$DoFinallyObserver.onComplete(CompletableDoFinally.java:78)
	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek$CompletableObserverImplementation.onComplete(CompletablePeek.java:115)
	at io.reactivex.rxjava3.internal.operators.single.SingleFlatMapCompletable$FlatMapCompletableObserver.onComplete(SingleFlatMapCompletable.java:102)
	at io.reactivex.rxjava3.internal.disposables.EmptyDisposable.complete(EmptyDisposable.java:68)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableEmpty.subscribeActual(CompletableEmpty.java:27)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.single.SingleFlatMapCompletable$FlatMapCompletableObserver.onSuccess(SingleFlatMapCompletable.java:91)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableAnySingle$AnySubscriber.onNext(FlowableAnySingle.java:89)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapSingle$ConcatMapSingleSubscriber.drain(FlowableConcatMapSingle.java:233)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapSingle$ConcatMapSingleSubscriber.request(FlowableConcatMapSingle.java:108)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableAnySingle$AnySubscriber.onSubscribe(FlowableAnySingle.java:66)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapSingle$ConcatMapSingleSubscriber.onSubscribeDownstream(FlowableConcatMapSingle.java:102)
	at io.reactivex.rxjava3.internal.operators.mixed.ConcatMapXMainSubscriber.onSubscribe(ConcatMapXMainSubscriber.java:75)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableFromIterable.subscribe(FlowableFromIterable.java:69)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableFromIterable.subscribeActual(FlowableFromIterable.java:47)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16144)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapSingle.subscribeActual(FlowableConcatMapSingle.java:59)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16144)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableAnySingle.subscribeActual(FlowableAnySingle.java:38)
	at io.reactivex.rxjava3.core.Single.subscribe(Single.java:4855)
	at io.reactivex.rxjava3.internal.operators.single.SingleFlatMapCompletable.subscribeActual(SingleFlatMapCompletable.java:44)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableDoFinally.subscribeActual(CompletableDoFinally.java:43)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableDefer.subscribeActual(CompletableDefer.java:43)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable$SourceObserver.onComplete(CompletableAndThenCompletable.java:67)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable$NextObserver.onComplete(CompletableAndThenCompletable.java:99)
	at io.reactivex.rxjava3.internal.util.AtomicThrowable.tryTerminateConsumer(AtomicThrowable.java:154)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapCompletable$ConcatMapCompletableObserver.drain(FlowableConcatMapCompletable.java:166)
	at io.reactivex.rxjava3.internal.operators.mixed.ConcatMapXMainSubscriber.onComplete(ConcatMapXMainSubscriber.java:123)
	at io.reactivex.rxjava3.internal.subscriptions.EmptySubscription.complete(EmptySubscription.java:69)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableFromIterable.subscribe(FlowableFromIterable.java:61)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableFromIterable.subscribeActual(FlowableFromIterable.java:47)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16144)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapCompletable.subscribeActual(FlowableConcatMapCompletable.java:56)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable$SourceObserver.onComplete(CompletableAndThenCompletable.java:67)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable$NextObserver.onComplete(CompletableAndThenCompletable.java:99)
	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek$CompletableObserverImplementation.onComplete(CompletablePeek.java:115)
	at io.reactivex.rxjava3.internal.util.AtomicThrowable.tryTerminateConsumer(AtomicThrowable.java:154)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapCompletable$ConcatMapCompletableObserver.drain(FlowableConcatMapCompletable.java:166)
	at io.reactivex.rxjava3.internal.operators.mixed.ConcatMapXMainSubscriber.onComplete(ConcatMapXMainSubscriber.java:123)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableDoOnEach$DoOnEachSubscriber.onComplete(FlowableDoOnEach.java:135)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableCache.replay(FlowableCache.java:245)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableCache.onComplete(FlowableCache.java:340)
	at io.reactivex.rxjava3.internal.util.AtomicThrowable.tryTerminateConsumer(AtomicThrowable.java:92)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableFlatMapMaybe$FlatMapMaybeSubscriber.drainLoop(FlowableFlatMapMaybe.java:311)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableFlatMapMaybe$FlatMapMaybeSubscriber.drain(FlowableFlatMapMaybe.java:269)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableFlatMapMaybe$FlatMapMaybeSubscriber.onComplete(FlowableFlatMapMaybe.java:149)
	at io.reactivex.rxjava3.internal.subscriptions.EmptySubscription.complete(EmptySubscription.java:69)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableEmpty.subscribeActual(FlowableEmpty.java:34)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16144)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableFlatMapMaybe.subscribeActual(FlowableFlatMapMaybe.java:53)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16144)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableCache.subscribeActual(FlowableCache.java:115)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16144)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16090)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableDefer.subscribeActual(FlowableDefer.java:42)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16144)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableDoOnEach.subscribeActual(FlowableDoOnEach.java:50)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16144)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapCompletable.subscribeActual(FlowableConcatMapCompletable.java:56)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable$SourceObserver.onComplete(CompletableAndThenCompletable.java:67)
	at io.reactivex.rxjava3.internal.util.AtomicThrowable.tryTerminateConsumer(AtomicThrowable.java:154)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapCompletable$ConcatMapCompletableObserver.drain(FlowableConcatMapCompletable.java:166)
	at io.reactivex.rxjava3.internal.operators.mixed.ConcatMapXMainSubscriber.onComplete(ConcatMapXMainSubscriber.java:123)
	at io.reactivex.rxjava3.internal.subscriptions.EmptySubscription.complete(EmptySubscription.java:69)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableFromIterable.subscribe(FlowableFromIterable.java:61)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableFromIterable.subscribeActual(FlowableFromIterable.java:47)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16144)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapCompletable.subscribeActual(FlowableConcatMapCompletable.java:56)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableResumeNext.subscribeActual(CompletableResumeNext.java:41)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableTimeout.subscribeActual(CompletableTimeout.java:53)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableDefer.subscribeActual(CompletableDefer.java:43)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableResumeNext.subscribeActual(CompletableResumeNext.java:41)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableResumeNext.subscribeActual(CompletableResumeNext.java:41)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.gravitee.gateway.reactive.handlers.api.v4.CompletableReactorChain.subscribeActual(CompletableReactorChain.java:86)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableDoFinally.subscribeActual(CompletableDoFinally.java:43)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableDefer.subscribeActual(CompletableDefer.java:43)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable$SourceObserver.onComplete(CompletableAndThenCompletable.java:67)
	at io.reactivex.rxjava3.internal.util.AtomicThrowable.tryTerminateConsumer(AtomicThrowable.java:154)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapCompletable$ConcatMapCompletableObserver.drain(FlowableConcatMapCompletable.java:166)
	at io.reactivex.rxjava3.internal.operators.mixed.ConcatMapXMainSubscriber.onSubscribe(ConcatMapXMainSubscriber.java:77)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableFromIterable.subscribe(FlowableFromIterable.java:69)
	at io.reactivex.rxjava3.internal.operators.flowable.FlowableFromIterable.subscribeActual(FlowableFromIterable.java:47)
	at io.reactivex.rxjava3.core.Flowable.subscribe(Flowable.java:16144)
	at io.reactivex.rxjava3.internal.operators.mixed.FlowableConcatMapCompletable.subscribeActual(FlowableConcatMapCompletable.java:56)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableDoFinally.subscribeActual(CompletableDoFinally.java:43)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletableResumeNext.subscribeActual(CompletableResumeNext.java:41)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.internal.operators.completable.CompletablePeek.subscribeActual(CompletablePeek.java:51)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2860)
	at io.reactivex.rxjava3.core.Completable.subscribe(Completable.java:2846)
	at io.gravitee.gateway.reactive.standalone.vertx.HttpProtocolVerticle.dispatchRequest(HttpProtocolVerticle.java:108)
	at io.gravitee.gateway.reactive.standalone.vertx.HttpProtocolVerticle.lambda$rxStart$4(HttpProtocolVerticle.java:82)
	at io.vertx.lang.rx.DelegatingHandler.handle(DelegatingHandler.java:20)
	at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:328)
	at io.vertx.core.impl.DuplicatedContext.emit(DuplicatedContext.java:166)
	at io.vertx.core.http.impl.Http1xServerConnection.handleMessage(Http1xServerConnection.java:174)
	at io.vertx.core.net.impl.ConnectionBase.read(ConnectionBase.java:159)
	at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:153)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.vertx.core.http.impl.Http1xUpgradeToH2CHandler.channelRead(Http1xUpgradeToH2CHandler.java:124)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.vertx.core.http.impl.Http1xOrH2CHandler.end(Http1xOrH2CHandler.java:61)
	at io.vertx.core.http.impl.Http1xOrH2CHandler.channelRead(Http1xOrH2CHandler.java:38)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:801)
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:509)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:407)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: io.gravitee.gateway.policy.PolicyException: java.lang.NullPointerException
	at io.gravitee.gateway.policy.impl.ExecutablePolicy.execute(ExecutablePolicy.java:113)
	at io.gravitee.gateway.reactive.policy.adapter.policy.PolicyAdapter.lambda$policyExecute$0(PolicyAdapter.java:102)
	... 243 common frames omitted
Caused by: java.lang.NullPointerException: null
```

Reply:
Now returning `-1` on error and then not including the limit if it is less than 0.